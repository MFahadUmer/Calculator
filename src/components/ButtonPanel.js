import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

const ButtonPanel = ({ clickHandler }) => {
  const buttonGroups = {
    groupOne: {
      id: 1,
      group: ['AC', '+/-', '%', '/'],
    },
    groupTwo: {
      id: 2,
      group: ['7', '8', '9', 'X'],
    },
    groupThree: {
      id: 3,
      group: ['4', '5', '6', '-'],
    },
    groupFour: {
      id: 4,
      group: ['1', '2', '3', '+'],
    },
    groupFive: {
      id: 5,
      group: ['0', '.', '='],
    },
  };

  return (
    <div>
      <div className="buttonPanel">
        {buttonGroups.groupOne.group.map(button => (
          <Button
            value={button}
            key={button}
            handleClick={() => clickHandler(button)}
          />
        ))}
      </div>
      <div className="buttonPanel">
        {buttonGroups.groupTwo.group.map(button => (
          <Button value={button} key={button} handleClick={() => clickHandler(button)} />
        ))}
      </div>
      <div className="buttonPanel">
        {buttonGroups.groupThree.group.map(button => (
          <Button value={button} key={button} handleClick={() => clickHandler(button)} />
        ))}
      </div>
      <div className="buttonPanel">
        {buttonGroups.groupFour.group.map(button => (
          <Button value={button} key={button} handleClick={() => clickHandler(button)} />
        ))}
      </div>
      <div className="buttonPanel">
        {buttonGroups.groupFive.group.map(button => (
          <Button
            wide
            value={button}
            key={button}
            handleClick={() => clickHandler(button)}
          />
        ))}
      </div>
    </div>
  );
};

ButtonPanel.propTypes = {
  clickHandler: PropTypes.func,
};

ButtonPanel.defaultProps = {
  clickHandler: '',
};

export default ButtonPanel;
