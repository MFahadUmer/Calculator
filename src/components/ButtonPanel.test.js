import React from 'react';
import renderer from 'react-test-renderer';
import ButtonPanel from './ButtonPanel';

describe('ButtonPanel Tests', () => {
  it('matches the buttonpanel component snapshot', () => {
    const tree = renderer.create(<ButtonPanel />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
