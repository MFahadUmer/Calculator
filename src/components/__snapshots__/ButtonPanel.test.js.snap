// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPanel Tests matches the buttonpanel component snapshot 1`] = `
<div>
  <div
    className="buttonPanel"
  >
    <button
      className="width25 button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="submit"
    >
      AC
    </button>
    <button
      className="width25 button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="submit"
    >
      +/-
    </button>
    <button
      className="width25 button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="submit"
    >
      %
    </button>
    <button
      className="width25 button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="submit"
    >
      /
    </button>
  </div>
  <div
    className="buttonPanel"
  >
    <button
      className="width25 button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="submit"
    >
      7
    </button>
    <button
      className="width25 button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="submit"
    >
      8
    </button>
    <button
      className="width25 button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="submit"
    >
      9
    </button>
    <button
      className="width25 button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="submit"
    >
      X
    </button>
  </div>
  <div
    className="buttonPanel"
  >
    <button
      className="width25 button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="submit"
    >
      4
    </button>
    <button
      className="width25 button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="submit"
    >
      5
    </button>
    <button
      className="width25 button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="submit"
    >
      6
    </button>
    <button
      className="width25 button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="submit"
    >
      -
    </button>
  </div>
  <div
    className="buttonPanel"
  >
    <button
      className="width25 button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="submit"
    >
      1
    </button>
    <button
      className="width25 button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="submit"
    >
      2
    </button>
    <button
      className="width25 button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="submit"
    >
      3
    </button>
    <button
      className="width25 button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="submit"
    >
      +
    </button>
  </div>
  <div
    className="buttonPanel"
  >
    <button
      className="width50 button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="submit"
    >
      0
    </button>
    <button
      className="width25 button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="submit"
    >
      .
    </button>
    <button
      className="width25 button"
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "orange",
        }
      }
      type="submit"
    >
      =
    </button>
  </div>
</div>
`;
